### PR TITLE
MONGOID-3917 create child documents as per _type passed

### PR DESCRIPTION
**Pull request by swapnilchincholkar**

### CODE REVIEW

**Comment by swapnilchincholkar** [comment]
Looks like 2 test cases failed, but it might not affected by my commit. Below are the details. Let me know if I need to do some changes in the my commit.

1.For mongodb 2.6.10
 Failure/Error: expect(Time.mongoize(time).to_f).to eq(time.to_time.to_f)
       
       expected: 1444993508.406
            got: 1444993508.405
       
       (compared using ==)
     # ./spec/mongoid/extensions/time_spec.rb:264:in `(root)'

2. For mongodb 3.0.3
Failure/Error: expect(Band.all.selector["likes"]).to_not eq(Band.all.selector["likes"])
       
       expected: value != {"$gt"=>53}
            got: {"$gt"=>53}
       
       (compared using ==)
       
       Diff:
       
     # ./spec/mongoid/scopable_spec.rb:81:in `(root)'

**Comment by johnnyshields** [comment]
@swapnilchincholkar can you rebase this PR on latest master and see if it's still failing?

**Comment by estolfo** [comment]
@swapnilchincholkar can you please rebase? thanks

File: lib/mongoid/relations/builders/nested_attributes/many.rb [hunk 1]
```diff
@@ -106,7 +106,8 @@ def process_attributes(parent, attrs)
                 update_document(doc, attrs)
               end
             else
-              existing.push(Factory.build(metadata.klass, attrs)) unless destroyable?(attrs)
+              klass = attrs[:_type].try(:constantize) || metadata.klass
+              existing.push(Factory.build(klass, attrs)) unless destroyable?(attrs)
```
Comment (johnnyshields) [inline lib/mongoid/relations/builders/nested_attributes/many.rb]: I think this code belongs in the Factory class actually. I will take a look

**Comment by estolfo** [comment]
This is fixed in Mongoid 7.0, see the ticket: https://jira.mongodb.org/browse/MONGOID-3917

**Pull request closed without merging**